Plugin errors when given file path with no root (happens for .java files)

Hi, @jrudolph 

I was looking into integrating your very neat plugin into a project of my own!
But I stumbled on the following error when it was run on my GH workflow:
```
Error:  Caused by: java.lang.IllegalArgumentException: 'other' is different type of Path
Error:  	at java.base/sun.nio.fs.UnixPath.relativize(UnixPath.java:406)
Error:  	at java.base/sun.nio.fs.UnixPath.relativize(UnixPath.java:51)
Error:  	at net.virtualvoid.hackersdigest.GithubActionCompileReporter.log(GithubActionCompileReporter.scala:22)
```
I have made a local copy of your plugin with some print statements above the erronous line:
```
println(baseDir.toPath)
println(position.sourceFile.get().toPath)
```
And I now see that the source file path for Java files do not have a root component.
This does not work with `relativize`.
> A relative path cannot be constructed if only one of the paths have a root component.

https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/nio/file/Path.html#relativize(java.nio.file.Path)

--- a/src/main/scala/net/virtualvoid/hackersdigest/GithubActionCompileReporter.scala
+++ b/src/main/scala/net/virtualvoid/hackersdigest/GithubActionCompileReporter.scala
@@ -15,9 +15,12 @@ class GithubActionCompileReporter(annotator: Annotator, delegate: xsbti.Reporter
     delegate.log(problem)
     import problem._
 
-    if ((severity == Severity.Warn || severity == Severity.Error) && position.sourceFile.isPresent) {
-      val file    = baseDir.toPath.relativize(position.sourceFile.get().toPath).toFile
-      val message = problem.message.split("\n").head
+    if (
+      (severity == Severity.Warn || severity == Severity.Error) && position.sourceFile.isPresent && position.sourceFile
+        .get()
+        .toPath
+        .getRoot != null
+    ) {
       def e(key: String, value: java.util.Optional[Integer]): String =
         value.map[String](v => s",$key=$v").orElse("")
 
@@ -27,7 +30,9 @@ class GithubActionCompileReporter(annotator: Annotator, delegate: xsbti.Reporter
         case _              => throw new IllegalStateException
       }
       // TODO: resurrect column info: ${e("col", position.startColumn())}${e("endColumn", position.endColumn())}
-      val line = if (position().line().isPresent) Some(position().line().get().intValue()) else None
+      val message = problem.message.split("\n").head
+      val file    = baseDir.toPath.relativize(position.sourceFile.get().toPath).toFile
+      val line    = if (position.line.isPresent) Some(position.line.get().intValue) else None
       annotator.createAnnotation(AnnotationOrigin.Compilation, level, message, Some(file.toString), line)
     }
   }

